lib/app.js

```diff
@@ -0,0 +1,67 @@
+const jwt = require('jsonwebtoken');
+const ui = require('@asl/service/ui');
+const { views, content } = require('@asl/pages');
+const db = require('@asl/schema');
+const Queue = require('./queue');
+
+module.exports = settings => {
+  const app = ui({
+    ...settings,
+    views,
+    content: {
+      siteTitle: 'Accept invitation'
+    }
+  });
+
+  const { Profile } = db(settings.db);
+
+  app.use(content);
+
+  app.use((req, res, next) => {
+    req.queue = Queue(settings.sqs);
+    next();
+  });
+
+  app.use((req, res, next) => {
+    res.locals.scripts = false;
+    next();
+  });
+
+  app.param('token', (req, res, next, token) => {
+    req.token = token;
+    next();
+  });
+
+  app.use('/:token', (req, res, next) => {
+    Promise.resolve()
+      .then(() => jwt.verify(req.token, settings.jwt.secret))
+      .then(model => {
+        req.model = {
+          ...model,
+          id: 'register-user'
+        };
+        next();
+      })
+      .catch(next);
+  });
```

Comment:
I'd probably just do all this lot in the `param` method.

Reply:
👍